sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_common.cpp

```diff
@@ -45,3 +45,214 @@ ur_result_t ze2urResult(ze_result_t ZeResult) {
   }
   return It->second;
 }
+
+void urPrint(const char *Format, ...) {
+  if (UrL0Debug & UR_L0_DEBUG_BASIC) {
+    va_list Args;
+    va_start(Args, Format);
+    vfprintf(stderr, Format, Args);
+    va_end(Args);
+  }
+}
+
+void zePrint(const char *Format, ...) {
+  if (ZeDebug & ZE_DEBUG_BASIC) {
+    va_list Args;
+    va_start(Args, Format);
+    vfprintf(stderr, Format, Args);
+    va_end(Args);
+  }
+}
+
+// This function will ensure compatibility with both Linux and Windows for
+// setting environment variables.
+bool setEnvVar(const char *name, const char *value) {
+#ifdef _WIN32
+  int Res = _putenv_s(name, value);
+#else
+  int Res = setenv(name, value, 1);
+#endif
+  if (Res != 0) {
+    zePrint(
```

Comment:
Seems we need a way to determine if this code is being executed as part of L0 PI or L0 UR adapter.   One way to do it would be to add a compile define into cmake that is only set when compiling the adapater library,  check it in this file and route calls accordingly.

Reply:
@bmyates: we could, but that would be short-termed since idea is that there's no PI+L0 anymore but PI+URL0, if I'm not mistaken. This is how the prints look like now with this change:

```
$ UR_L0_DEBUG=1 SYCL_PI_TRACE=2 <app>
...
---> piKernelRelease(
        <unknown> : 0x2a78f10
ZE ---> zeKernelDestroy(Kernel->ZeKernel)
UR ---> urProgramRelease(KernelProgram)
) --->  pi_result : PI_SUCCESS
...
```